scripts/cd/olm-setup.sh

```diff
@@ -8,6 +8,7 @@ user_help () {
     echo "options:"
     echo "-pr, --project-root      Path to the root of the project the CSV should be generated for/in"
     echo "-nv, --next-version      Semantic version of the new CSV to be created"
+    echo "-rv, --replace-version   Semantic version of the CSV to be replaced (without the operator name prefix - eg: 0.0.1035-460-commit-e2d2b30-280e4ec)"
```

Comment:
If you mention `semantic`, it is good practice to validate the input using a regular expression (or another method) to ensure it meets the expected format (semver, or something custom or more relaxed). If you don't, I would avoid using the word `semantic`, as -- per implementation -- you accept any string basically.

Reply:
in theory yes. If this is a generic script to be used by people outside of our team, sure why not. But this is not the case. In this specific case, it's expected that you know what you do; otherwise, you can break many things there.

Apart from that. 
* This script is not supposed to be used directly on its own - it should be used only via the makefile targets in the repositories
* operator-sdk/kubebuilder (or opm, not sure who does that) has it's own validation for the semantic format.

Anyway, I can drop the word :-) 
```suggestion
    echo "-rv, --replace-version   The CSV version to be replaced (without the operator name prefix - eg: 0.0.1035-460-commit-e2d2b30-280e4ec)"
```

Reply:
my point is, let's keep the logic simple, there is no need for any validation because you really have to know what you are doing when running the CD process directly/manually

Reply:
+1  -- yup that's totally fine, let's keep the logic simple (not for the reasons you mentioned though, mistakes can always happen even if you think you know what you are doing). I agree not every script needs to be perfect and fail-check everything, actually most of them are dumb on purpose, so let's keep it simple and avoid regex hell ;)

However, we need to at least make sure this variable is populated at some point of the CI/CD pipeline. If not in this script, then in the makefile or somewhere else. It's very usual to end up with `--replace-version=""` depending on the ENV and config you are supposed to run this.

Reply:
see the other PRs:
https://github.com/codeready-toolchain/host-operator/pull/1134
https://github.com/codeready-toolchain/registration-service/pull/503

and I'll create a doc describing how to use it

Reply:
Díky moc Matous! I am resolving the conversation :)